test/libyul/Common.cpp

```diff
@@ -96,7 +96,9 @@ yul::Block yul::test::disambiguate(std::string const& _source, bool _yul)
 
 std::string yul::test::format(std::string const& _source, bool _yul)
 {
-	return yul::AsmPrinter()(*parse(_source, _yul).first);
+	auto const version = solidity::test::CommonOptions::get().evmVersion();
+	Dialect const& dialect = _yul ? EVMDialectTyped::instance(version) : EVMDialect::strictAssemblyForEVMObjects(version);
```

Comment:
Isn't it probably enough to just use the dialect? E.g. `EVMDialectTyped` will behave like `FullTypeInformation`, if used within `AsmPrinter`. Where if dialect was `EVMDialect` the printing behaves like `OmitDefaultType`. Do we have cases where we will use a `EVMDialectTyped` dialect where we don't want to print the type information and e.g. vice versa, if we use `EVMDialect` that we want to print `FullTypeInformation`? I'm just asking, so far I didn't really check how the dialect mechanism exactly work.

Reply:
> Is this really correct?

Previously the dialect came from the `parse(_source, _yul)` call, hidden in the `YulStack`. If 
- `_yul == true`, then the stack gets a `YulStack::Language::Yul`, which results in a `EVMDialectTyped::instance(_version)`;
- `_yul == false`, the stack gets a `YulStack::Language::StrictAssembly`, which results in `EVMDialect::strictAssemblyForEVMObjects`. 

The conversion happens [here](https://github.com/ethereum/solidity/blob/35231a1b34cd4fe53045c590b7a6eda2935ae9ee/libyul/YulStack.cpp#L51). 

> Isn't it probably to just use the dialect? 

I don't think so, see, e.g., [here](https://github.com/ethereum/solidity/blob/96fdcc3d040d05209f561e78a50dee2ebe4252c3/libsolidity/codegen/ir/IRGeneratorForStatements.cpp#L2260). This will always print type info, irrespective of the dialect. If the dialect had a default type for example and the dialect was provided here as argument, it wouldn't be printed.